test/PackUnPackTest.cc

```diff
@@ -0,0 +1,874 @@
+                             AreaSize(16, 8),
+                             AreaSize(8, 16),
+                             AreaSize(32, 32),
+                             AreaSize(32, 8),
+                             AreaSize(16, 32),
+                             AreaSize(8, 32),
+                             AreaSize(32, 16),
+                             AreaSize(16, 64),
+                             AreaSize(64, 16),
+                             AreaSize(64, 64),
+                             AreaSize(64, 32),
+                             AreaSize(32, 64)};
+typedef std::tuple<AreaSize, TestPattern> TestParam;
+
+/**
+ * @brief Unit test for Pack functions include:
+ *  - c_pack_avx2_intrin
+ *
+ * Test strategy:
+ *  This test case combine different pu width{32-64} x search area
+ * height{32-64} and different test pattern(REF_MIN, REF_MAX, REF_RANDOM). Check
+ * the result by compare result from reference function and avx2 function.
+ *
+ *
+ * Expect result:
+ *  Results from reference function and avx2 function are
+ * equal.
+ *
+ *
+ * Test cases:
+ *  Width {32, 64} x height{32, 64}
+ *  Test vector pattern {REF_MIN, REF_MIN, REF_RANDOM}
+ *
+ */
+class PackTest : public ::testing::Test,
+                 public ::testing::WithParamInterface<TestParam> {
+  public:
+    PackTest()
+        : test_pattern_(TEST_GET_PARAM(1)),
+          area_width_(std::get<0>(TEST_GET_PARAM(0))),
+          area_height_(std::get<1>(TEST_GET_PARAM(0))) {
+        in_stride_ = out_stride_ = MAX_SB_SIZE;
+        test_size_ = MAX_SB_SQUARE;
+    }
+
+    void SetUp() {
+        in_compn_bit_buffer1_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        in_compn_bit_buffer2_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        in_bit_buffer_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        local_cache1_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        local_cache2_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        memset(in_compn_bit_buffer1_,
+               0,
+               test_size_ * sizeof(in_compn_bit_buffer1_[0]));
+        memset(in_compn_bit_buffer2_,
+               0,
+               test_size_ * sizeof(in_compn_bit_buffer2_[0]));
+        memset(local_cache1_, 0, test_size_ * sizeof(local_cache1_[0]));
+        memset(local_cache2_, 0, test_size_ * sizeof(local_cache2_[0]));
+    }
+
+    void TearDown() override {
+        if (in_bit_buffer_)
+            eb_aom_free(in_bit_buffer_);
+        if (in_compn_bit_buffer1_)
+            eb_aom_free(in_compn_bit_buffer1_);
+        if (in_compn_bit_buffer2_)
+            eb_aom_free(in_compn_bit_buffer2_);
+        if (local_cache1_)
+            eb_aom_free(local_cache1_);
+        if (local_cache2_)
+            eb_aom_free(local_cache2_);
+    }
+
+  protected:
+    void prepare_data() {
```

Comment:
From my point of view, the target of the pack/unpack operators could be the picture instead of PU or search area, so maybe the input width/height should not limited to the AreaSize.